tests/api/api/test_datastore_profiles.py

```diff
@@ -0,0 +1,191 @@
+        ),
+    )
+    resp = client.post(legacy_api_projects_path, json=project.dict())
+    assert resp.status_code == HTTPStatus.CREATED.value
+    return resp
+
+
+def test_datastore_profile_create_ok(db: Session, client: TestClient):
+    _create_project(client)
+    resp = client.post(
+        api_datastore_path.format(project=project),
+        data=json.dumps(datastore),
+    )
+    assert resp.status_code == HTTPStatus.OK.value
+
+    expected_return = {"project": project, **datastore}
+
+    resp = client.get(
+        api_datastore_path.format(project=project) + "/" + datastore["name"],
+    )
+    assert resp.status_code == HTTPStatus.OK.value
+    assert json.loads(resp._content) == expected_return
+
+
+def test_datastore_profile_update_ok(db: Session, client: TestClient):
+    _create_project(client)
+    resp = client.post(
+        api_datastore_path.format(project=project),
+        data=json.dumps(datastore),
+    )
+    assert resp.status_code == HTTPStatus.OK.value
+    datastore_updated = datastore
+    datastore_updated["body"] = "another version of body"
+    resp = client.post(
+        api_datastore_path.format(project=project),
+        data=json.dumps(datastore_updated),
+    )
+    assert resp.status_code == HTTPStatus.OK.value
+
+    expected_return = {"project": project, **datastore_updated}
+
+    resp = client.get(
+        api_datastore_path.format(project=project) + "/" + datastore["name"],
+    )
+    assert resp.status_code == HTTPStatus.OK.value
+    assert json.loads(resp._content) == expected_return
+
+
+def test_datastore_profile_create_fail(db: Session, client: TestClient):
+    # No project created
+    resp = client.post(
+        api_datastore_path.format(project=project),
+        data=json.dumps(datastore),
+    )
+    assert resp.status_code == HTTPStatus.NOT_FOUND.value
+
+    # Empty data
+    _create_project(client)
+    resp = client.post(
+        api_datastore_path.format(project=project),
+        data={},
+    )
+    assert resp.status_code == HTTPStatus.UNPROCESSABLE_ENTITY.value
+
+
+def test_datastore_profile_get_fail(db: Session, client: TestClient):
+    # No project created
+    resp = client.get(
+        api_datastore_path.format(project=project) + "/" + datastore["name"],
+    )
+    assert resp.status_code == HTTPStatus.NOT_FOUND.value
+
+    # Not existing profile
+    _create_project(client)
+    resp = client.post(
+        api_datastore_path.format(project=project),
+        data={},
+    )
+    resp = client.get(
+        api_datastore_path.format(project=project) + "/invalid",
+    )
+    assert resp.status_code == HTTPStatus.NOT_FOUND.value
+
+
+def test_datastore_profile_delete(db: Session, client: TestClient):
+    # No project created
+    resp = client.delete(
+        api_datastore_path.format(project=project) + "/" + datastore["name"],
+    )
+    assert resp.status_code == HTTPStatus.NOT_FOUND.value
```

Comment:
Might seem  a bit excessive, but I think each case should have its own test, so that one failure does not stop other test cases from running.

Reply:
done